File: 8.maximum-subarray/MaximumSubArray.py
def maxSubArray(nums):
    if len(nums) == 0:
        return 0
    result = nums[0]
    sum = 0
    for num in nums:
        if(sum < 0):
            sum = num
        else:
            sum += num
        result = max(result, sum)
    return result

if __name__ == "__main__":
    nums = [1, 2, 3, 4, 5]
    print(maxSubArray(nums))

File: 8.maximum-subarray/MaximumSubArray.java

class MaximumSubArray {

	public int maxSubArray(int[] nums) {
		if(nums.length == 0) {
			return 0;
		}
		int result = nums[0];
		int sum = 0;
		for (int num: nums) {
			if(sum < 0) {
				sum = num;
			} else {
				sum += num;
			}
			result = Math.max(result, sum);
		}
		return result;
	}

	public static void main(String[] args) {
		int[] nums = {1, -2,2,3, -1,4};
		System.out.println(new MaximumSubArray().maxSubArray(nums));
	}
}
